src/main/java/org/craftercms/engine/service/context/SiteContext.java

```diff
@@ -341,6 +351,7 @@ public void init(boolean waitTillFinished) throws SiteContextInitializationExcep
                     logger.info("--------------------------------------------------");
 
                     publishEvent(new SiteContextInitializedEvent(this));
+                    initializationLatch.countDown();
```

Comment:
What happens if the init fails? shouldn't this be in the finally? that way even if it fails it won't get other threads stuck.